snap/snapcraft.yaml

```diff
@@ -0,0 +1,118 @@
+name: edgex-device-onvif-camera
+base: core20
+license: Apache-2.0
+adopt-info: metadata
+
+architectures:
+  - build-on: amd64
+  - build-on: arm64
+
+grade: stable
+confinement: strict
+
+slots:
+  edgex-secretstore-token:
+    interface: content
+    source:
+      write: [$SNAP_DATA/device-onvif-camera]
+
+plugs:
+  device-config:
+    interface: content
+    target: $SNAP_DATA/config/device-onvif-camera
+
+apps:
+  device-onvif-camera:
+    adapter: full
+    command: bin/device-onvif-camera -confdir $SNAP_DATA/config/device-onvif-camera -profile res --registry $CONSUL_ADDR
```

Comment:
1. What is `-profile res` for?
2. `--registry` doesn't take any value: https://docs.edgexfoundry.org/2.2/microservices/configuration/CommonCommandLineOptions/#registry
3. Address to config provider should be passed as `--cp=consul.http://localhost:8500`.